test/lendingPool.js

```diff
@@ -1,103 +1,184 @@
+        await expect(this.lendingPool.connect(this.oracle).deposit({ value: ONE_ETH })).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("allows owner to add liquidators", async function() {
+        await this.lendingPool.connect(this.owner).addLiquidator(this.liquidator.address)
+    })
+
+    it ("blocks non-owners from adding liquidators", async function() {
+        await expect(this.lendingPool.connect(this.oracle).addLiquidator(this.liquidator.address)).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("blocks non-owners from borrowing NFTs", async function() {
+        const signature = await sign(this.oracle, PRICE, DEADLINE, this.nft.address);
+        await expect(this.lendingPool.connect(this.owner).borrow([683971, 683972], PRICE, DEADLINE, signature.v, signature.r, signature.s)).to.be.revertedWith("not owner");
+    });
+
+    it("blocks users from borrowing the same NFT twice", async function() {
+        const signature = await sign(this.oracle, PRICE, DEADLINE, this.nft.address);
+        // N.B.: baked into this test is the fact that `user` has already borrowed on Goerli at fork-time
+        await this.nft.connect(this.user).setApprovalForAll(this.lendingPool.address, true);
+
+        await expect(this.lendingPool.connect(this.user).borrow([683971, 683971], PRICE, DEADLINE, signature.v, signature.r, signature.s)).to.be.revertedWith("not owner");
+    });
+
+    it("sends eth to the user upon borrowing their NFTs", async function() {
+        const signature = await sign(this.oracle, PRICE, DEADLINE, this.nft.address);
+        const prevEth = await ethers.provider.getBalance(userAddress);
+        // N.B.: baked into this test is the fact that `user` has already borrowed on Goerli at fork-time
+        const pendingTx = await this.lendingPool.connect(this.user).borrow([683971, 683972], PRICE, DEADLINE, signature.v, signature.r, signature.s);
+        
+        const tx = await pendingTx.wait();
+
+        const postEth = await ethers.provider.getBalance(userAddress);
+
+        expect(await this.nft.ownerOf(683971)).to.equal(this.lendingPool.address)
+
+        expect(postEth.sub(prevEth)).to.be.equal(ethers.BigNumber.from(PRICE).mul(2).sub(tx.gasUsed * tx.effectiveGasPrice))
+    });
+
+    it("prevents non-owners from repaying loans", async function() {
+        await expect(this.lendingPool.connect(this.owner).repay([1], { value: (Number(ONE_TENTH_OF_AN_ETH) * 2).toFixed(0) })).to.be.revertedWith("not owner");
+    });
+
+    it("returns a correct tokenURI", async function () {
+        expect(await this.lendingPool.tokenURI(1)).to.eq(`https://nft.llamalend.com/nft/31337/${this.lendingPool.address.toLowerCase()}/0xf5de760f2e916647fd766b4ad9e85ff943ce3a2b/1`)
+    });
+
+    it("returns correct interest rates", async function () {
+        expect(Number(await this.lendingPool.currentAnnualInterest(0))).to.be.approximately(0.16e18, 200000)
```

Comment:
does this work? this only works in original tests because the amount borrowed matches

Reply:
yea they're all still sequential so should be the same behaviour